src/mame/video/toaplan_scu.h

```diff
@@ -6,22 +6,24 @@
 
 #pragma once
 
-
-class toaplan_scu_device : public device_t, public device_gfx_interface
+class toaplan_scu_device : public device_t, public device_gfx_interface, public device_video_interface
 {
 public:
 	toaplan_scu_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
 
+	typedef device_delegate<void (u8 priority, u32 &pri_mask)> pri_cb_delegate;
+
 	// configuration
 	void set_xoffsets(int xoffs, int xoffs_flipped)
 	{
 		m_xoffs = xoffs;
 		m_xoffs_flipped = xoffs_flipped;
 	}
+	template <typename... T> void set_pri_callback(T &&... args) { m_pri_cb.set(std::forward<T>(args)...); }
 
-	void draw_sprites_to_tempbitmap(const rectangle &cliprect, u16* spriteram, u32 bytes);
-	void copy_sprites_from_tempbitmap(bitmap_rgb32 &bitmap, const rectangle &cliprect, int priority);
-	void alloc_sprite_bitmap(screen_device &screen);
+	template<class BitmapClass> void draw_sprites_common(BitmapClass &bitmap, const rectangle &cliprect, u16* spriteram, u32 bytes);
```

Comment:
Are you expecting this to be directly used from outside the class?  If not, it should be protected or private.